pkg/manager/internal.go

```diff
@@ -420,27 +402,40 @@ func (cm *controllerManager) serveHealthProbes(stop <-chan struct{}) {
 		Handler: mux,
 	}
 	// Run server
-	go func() {
+	cm.startRunnable(RunnableFunc(func(stop <-chan struct{}) error {
 		if err := server.Serve(cm.healthProbeListener); err != nil && err != http.ErrServerClosed {
-			cm.errSignal.SignalError(err)
+			return err
 		}
-	}()
+		return nil
+	}))
 	cm.healthzStarted = true
 	cm.mu.Unlock()
 
 	// Shutdown the server when stop is closed
 	<-stop
 	if err := server.Shutdown(context.Background()); err != nil {
-		cm.errSignal.SignalError(err)
+		cm.errChan <- err
 	}
 }
 
-func (cm *controllerManager) Start(stop <-chan struct{}) error {
-	// join the passed-in stop channel as an upstream feeding into cm.internalStopper
-	defer close(cm.internalStopper)
+func (cm *controllerManager) Start(stop <-chan struct{}) (err error) {
+	// This chan indicates that stop is complete, in other words all runnables have returned or timeout on stop request
+	stopComplete := make(chan struct{})
+	defer close(stopComplete)
+	// This must be deferred after closing stopComplete, otherwise we deadlock
+	defer func() {
+		stopErr := cm.engageStopProcedure(stopComplete)
```

Comment:
very important: link to a picture of picard

Reply:
hmpf. I usually use references to (non scifi,fantasy) literature characters where possible to falsify the cliche of "all it ppl are somehow in Star Trek, Star Wars or LOTR". But I don't know of a commander off-hand

Reply:
(That approach is also very helpful in quickly identifying which code I wrote vs which code others wrote :upside_down_face: )